ann_benchmarks/runner.py

```diff
@@ -108,42 +110,73 @@ def run(definition, dataset, count, run_count, batch):
     X_test = numpy.array(D['test'])
     distance = D.attrs['distance']
     print('got a train set of size (%d * %d)' % (X_train.shape[0], dimension))
-    print('got %d queries' % len(X_test))
 
     X_train, X_test = dataset_transform(D)
+    # X_train = X_train[:5000]
 
     try:
         prepared_queries = False
         if hasattr(algo, "supports_prepared_queries"):
             prepared_queries = algo.supports_prepared_queries()
 
-        t0 = time.time()
-        memory_usage_before = algo.get_memory_usage()
-        algo.fit(X_train)
-        build_time = time.time() - t0
-        index_size = algo.get_memory_usage() - memory_usage_before
-        print('Built index in', build_time)
-        print('Index size: ', index_size)
+        if not test_only:
+            per_client = len(X_train) // num_clients
+            offset = per_client * (id - 1)
+            if num_clients == id:
+                fit_args = [X_train[offset:]]
+            else:
+                fit_args = [X_train[offset:(offset + per_client)]]
+            print('inserting %d out of %d vectors' % (len(fit_args[0]), len(X_train)))
+            if "offset" and "cap" in inspect.getfullargspec(algo.fit)[0]:
+                fit_args.append(offset)
+                fit_args.append(len(X_train))
+            
+            t0 = time.time()
+            memory_usage_before = algo.get_memory_usage()
```

Comment:
do we support this command?

Reply:
No. the default is getting the process memory usage, which of course is meaningless, but having it asking the server its usage will give mixed results of all clients, so it is best to leave this outside client scope as well.

Reply:
so lets have the right command here